### ISSUE TITLE
PostgreSQL Target Fails Assert Error

### ISSUE THREAD

**Issue by TheCedarPrince** [OHDSI/SqlRender#279]
I am working with DatabaseConnector and get the following error:

``` r
library("DatabaseConnector")
library("Capr")
#> Loading required package: CirceR
#> Loading required package: magrittr
connectionDetails <- createConnectionDetails(dbms = "postgresql", server = "[redacted]synpuf_v5", user = "synpuf_v5", password = "[redacted]", port = "5434", pathToDriver="[redacted]drivers")
connection <- connect(connectionDetails)
#> Connecting using PostgreSQL driver
renderTranslateQuerySql(connection, "SELECT * FROM person LI
MIT 1;")
#> Error in tryCatch(withCallingHandlers({: 1 assertions failed:
#>  * Variable 'targetDialect': Must be element of set {'sql
#>  * server','oracle','postgresql','pdw','impala','netezza','bigquery','spark','sqlite','redshift','hive','sqlite
#>  * extended'}, but types do not match (character != factor).
```

<sup>Created on 2022-07-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

It would appear that there is some sort of transitive error happening but I am not sure what it is.
Here is my versioninfo for R:

```
R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS/LAPACK: /Users/jzelko3/.julia/conda/3/lib/R/lib/libRblas.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] SqlRender_1.9.1         Capr_1.0.3              magrittr_2.0.3
[4] DatabaseConnector_5.0.4 CirceR_1.1.1            reprex_2.0.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     compiler_3.6.1   pillar_1.7.0     highr_0.9
 [5] tools_3.6.1      uuid_1.1-0       digest_0.6.18    bit_4.0.4
 [9] jsonlite_1.8.0   tibble_3.1.7     evaluate_0.15    lifecycle_1.0.1
[13] pkgconfig_2.0.3  rlang_1.0.3      DBI_1.1.3        cli_3.3.0
[17] rstudioapi_0.10  yaml_2.2.0       xfun_0.31        fastmap_1.1.0
[21] rJava_1.0-6      withr_2.5.0      dplyr_1.0.9      knitr_1.39
[25] generics_0.1.2   fs_1.2.7         vctrs_0.4.1      tidyselect_1.1.2
[29] bit64_4.0.5      glue_1.6.2       R6_2.5.1         processx_3.6.1
[33] fansi_1.0.3      rmarkdown_2.14   RJSONIO_1.3-1.6  purrr_0.3.4
[37] callr_3.7.0      clipr_0.8.0      ps_1.3.0         htmltools_0.5.2
[41] ellipsis_0.3.2   utf8_1.2.2       crayon_1.5.1       
```

Any thoughts as to what is happening? 
Thank you!

**Comment by schuemie** [comment]
I'm pretty sure this is because R versions older than 4 did auto-casting of strings to factors. I just pushed a [commit](https://github.com/OHDSI/SqlRender/commit/4053c09ea26d55c2631058f17a49b900a0bdf746) to the SqlRender develop branch that should fix this.

But overall I highly recommend upgrading to R v 4. Other HADES packages, like ParallelLogger, will not work on R < 4. (I did [ask](https://forums.ohdsi.org/t/requiring-r-v4/15563/2) if people were ok with requiring R 4 ;-) )
